### ISSUE TITLE
AdminType tabs don't work inside CollectionTypes as of 3.38.0

### ISSUE THREAD

**Issue by c0ntax** [sonata-project/SonataAdminBundle#5306]
#### Sonata packages

```
sonata-project/admin-bundle              3.40.3 3.40.3 The missing Symfony Admin Generator
sonata-project/block-bundle              3.12.1 3.12.1 Symfony SonataBlockBundle
sonata-project/cache                     2.0.1  2.0.1  Cache library
sonata-project/core-bundle               3.12.0 3.12.0 Symfony SonataCoreBundle
sonata-project/datagrid-bundle           2.4.0  2.4.0  Symfony SonataDatagridBundle
sonata-project/doctrine-extensions       1.1.0  1.1.0  Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle 3.6.3  3.6.3  Symfony Sonata / Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/exporter                  1.9.1  1.9.1  Lightweight Exporter library
```

#### Symfony packages

```
symfony/asset                v3.4.18 v3.4.18 Symfony Asset Component
symfony/cache                v3.4.18 v3.4.18 Symfony Cache component with PSR-6, PSR-16, and tags
symfony/class-loader         v3.4.18 v3.4.18 Symfony ClassLoader Component
symfony/config               v3.4.18 v3.4.18 Symfony Config Component
symfony/console              v3.4.18 v3.4.18 Symfony Console Component
symfony/debug                v3.4.18 v3.4.18 Symfony Debug Component
symfony/dependency-injection v3.4.18 v3.4.18 Symfony DependencyInjection Component
symfony/doctrine-bridge      v3.4.18 v3.4.18 Symfony Doctrine Bridge
symfony/dotenv               v3.4.18 v3.4.18 Registers environment variables from a .env file
symfony/event-dispatcher     v3.4.18 v3.4.18 Symfony EventDispatcher Component
symfony/expression-language  v3.4.18 v3.4.18 Symfony ExpressionLanguage Component
symfony/filesystem           v3.4.18 v3.4.18 Symfony Filesystem Component
symfony/finder               v3.4.18 v3.4.18 Symfony Finder Component
symfony/flex                 v1.1.8  v1.1.8  Composer plugin for Symfony
symfony/form                 v3.4.18 v3.4.18 Symfony Form Component
symfony/framework-bundle     v3.4.18 v3.4.18 Symfony FrameworkBundle
symfony/http-foundation      v3.4.18 v3.4.18 Symfony HttpFoundation Component
symfony/http-kernel          v3.4.18 v3.4.18 Symfony HttpKernel Component
symfony/inflector            v3.4.18 v3.4.18 Symfony Inflector Component
symfony/intl                 v3.4.18 v3.4.18 A PHP replacement layer for the C intl extension that includes additional data from the ICU lib...
symfony/maker-bundle         v1.9.0  v1.9.0  Symfony Maker helps you create empty commands, controllers, form classes, tests and more so you...
symfony/options-resolver     v3.4.18 v3.4.18 Symfony OptionsResolver Component
symfony/orm-pack             v1.0.5  v1.0.5  A pack for the Doctrine ORM
symfony/polyfill-apcu        v1.10.0 v1.10.0 Symfony polyfill backporting apcu_* functions to lower PHP versions
symfony/polyfill-intl-icu    v1.10.0 v1.10.0 Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-mbstring    v1.10.0 v1.10.0 Symfony polyfill for the Mbstring extension
symfony/process              v3.4.18 v3.4.18 Symfony Process Component
symfony/property-access      v3.4.18 v3.4.18 Symfony PropertyAccess Component
symfony/routing              v3.4.18 v3.4.18 Symfony Routing Component
symfony/security             v3.4.18 v3.4.18 Symfony Security Component
symfony/security-acl         v3.0.1  v3.0.1  Symfony Security Component - ACL (Access Control List)
symfony/security-bundle      v3.4.18 v3.4.18 Symfony SecurityBundle
symfony/templating           v3.4.18 v3.4.18 Symfony Templating Component
symfony/translation          v3.4.18 v3.4.18 Symfony Translation Component
symfony/twig-bridge          v3.4.18 v3.4.18 Symfony Twig Bridge
symfony/twig-bundle          v3.4.18 v3.4.18 Symfony TwigBundle
symfony/validator            v3.4.18 v3.4.18 Symfony Validator Component
symfony/web-server-bundle    v3.4.18 v3.4.18 Symfony WebServerBundle
symfony/yaml                 v3.4.18 v3.4.18 Symfony Yaml Component
```

#### PHP version

```
PHP 7.2.12-1+ubuntu18.04.1+deb.sury.org+1 (cli) (built: Nov 12 2018 09:55:44) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.12-1+ubuntu18.04.1+deb.sury.org+1, Copyright (c) 1999-2018, by Zend Technologies
    with Xdebug v2.6.1, Copyright (c) 2002-2018, by Derick Rethans
```

## Subject

AdminType form elements' tab ids are incorrect.

## Steps to reproduce

1. Unzip [sonata-test.tar.gz](https://github.com/sonata-project/SonataAdminBundle/files/2604660/sonata-test.tar.gz)
2. `cd sonata-test`
3. `composer install`
4. `bin/console server:run`
5. Navigate to http://127.0.0.1:8000/admin/app/town/create
6. Add a town name
7. Click `Add a street`
8. Add a street name
9. Click 'Add a building'
10. Fill in anything for the Address and Occupant tabs
11. Click on 'Add a street' again
12. Click on 'Add a building' for the new street
13. Click on the Tab heading for either Address or Occupant

## Expected results

The correct Address or Occupant tab is switched to in the second Street/Building group

## Actual results

The Address or Occupant is switched in the first Street/Building group

**Comment by kunicmarko20** [comment]
could you do a `git bisect`? to determine which commit broke it?

**Comment by c0ntax** [comment]
```
24ead937eee2f869f1a2848b0110b0a38b806cf3 is the first bad commit
commit 24ead937eee2f869f1a2848b0110b0a38b806cf3
Author: Emmanuel Vella <vella.emmanuel@gmail.com>
Date:   Tue Aug 7 19:12:22 2018 +0200

    Use consistent uniqids (#5176)

:040000 040000 c6e77bb92ca19dacf5cd2477d7a3c9f761f00036 045d7c4e2faddff4e4116a9535082c1d9835fec6 M      src
:040000 040000 e290008d92c28938a46d45aea51f77bb11f139eb d81e90d36f2849f4d01cf6371356e1b4b16a6467 M      tests
```

**Comment by kunicmarko20** [comment]
https://github.com/sonata-project/SonataAdminBundle/pull/5176

@EmmanuelVella could you maybe help with the issue

**Comment by c0ntax** [comment]
I'm guessing that the issue lies with the use of `$this->getBaseCodeRoute()` in this commit:

https://github.com/sonata-project/SonataAdminBundle/commit/24ead937eee2f869f1a2848b0110b0a38b806cf3#diff-61e6d1f8fafe87b868adbe28ce0ba8f6R1869

Sadly I don't know enough about the internals of Sonata to know what to change it to

**Comment by EmmanuelVella** [comment]
Hello,

Sorry to hear my PR broke this. Does this commit fixes it ? https://github.com/EmmanuelVella/SonataAdminBundle/commit/4f176558830010655e1c6b78005aef78e74dd20f

If yes, I will do a new PR.

**Comment by c0ntax** [comment]
I have patched 3.40.3 and can confirm that this fixes the issue. :-)
